icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java

```diff
@@ -334,75 +331,28 @@ protected ULocale createInstance(Locale key, Void unused) {
-        { "en_BE_PREEURO",  "en_BE", "currency", "BEF" },
-        { "en_IE_PREEURO",  "en_IE", "currency", "IEP" },
-        { "es__TRADITIONAL", "es", "collation", "traditional" }, /* Old ICU name */
-        { "es_ES_PREEURO",  "es_ES", "currency", "ESP" },
-        { "eu_ES_PREEURO",  "eu_ES", "currency", "ESP" },
-        { "fi_FI_PREEURO",  "fi_FI", "currency", "FIM" },
-        { "fr_BE_PREEURO",  "fr_BE", "currency", "BEF" },
-        { "fr_FR_PREEURO",  "fr_FR", "currency", "FRF" },
-        { "fr_LU_PREEURO",  "fr_LU", "currency", "LUF" },
-        { "ga_IE_PREEURO",  "ga_IE", "currency", "IEP" },
-        { "gl_ES_PREEURO",  "gl_ES", "currency", "ESP" },
-        { "hi__DIRECT",     "hi", "collation", "direct" }, /* Old ICU name */
-        { "it_IT_PREEURO",  "it_IT", "currency", "ITL" },
-        { "ja_JP_TRADITIONAL", "ja_JP", "calendar", "japanese" },
-      //{ "nb_NO_NY",       "nn_NO", null, null },
-        { "nl_BE_PREEURO",  "nl_BE", "currency", "BEF" },
-        { "nl_NL_PREEURO",  "nl_NL", "currency", "NLG" },
-        { "pt_PT_PREEURO",  "pt_PT", "currency", "PTE" },
-        { "sl_ROZAJ",       "sl__ROZAJ", null, null }, /* registered name */
-        { "sr_SP_CYRL",     "sr_Cyrl_RS", null, null }, /* .NET name */
-        { "sr_SP_LATN",     "sr_Latn_RS", null, null }, /* .NET name */
-        { "sr_YU_CYRILLIC", "sr_Cyrl_RS", null, null }, /* Linux name */
-        { "th_TH_TRADITIONAL", "th_TH", "calendar", "buddhist" }, /* Old ICU name */
-        { "uz_UZ_CYRILLIC", "uz_Cyrl_UZ", null, null }, /* Linux name */
-        { "uz_UZ_CYRL",     "uz_Cyrl_UZ", null, null }, /* .NET name */
-        { "uz_UZ_LATN",     "uz_Latn_UZ", null, null }, /* .NET name */
-        { "zh_CHS",         "zh_Hans", null, null }, /* .NET name */
-        { "zh_CHT",         "zh_Hant", null, null }, /* .NET name */
-        { "zh_GAN",         "zh__GAN", null, null }, /* registered name */
-        { "zh_GUOYU",       "zh", null, null }, /* registered name */
-        { "zh_HAKKA",       "zh__HAKKA", null, null }, /* registered name */
-        { "zh_MIN",         "zh__MIN", null, null }, /* registered name */
-        { "zh_MIN_NAN",     "zh__MINNAN", null, null }, /* registered name */
-        { "zh_WUU",         "zh__WUU", null, null }, /* registered name */
-        { "zh_XIANG",       "zh__XIANG", null, null }, /* registered name */
-        { "zh_YUE",         "zh__YUE", null, null } /* registered name */
-    };
-
-    /**
-     * This table lists pairs of locale ids for canonicalization.
-     * The first item is the normalized variant id.
-     */
-    private static String[][] variantsToKeywords = {
-        { "EURO",   "currency", "EUR" },
-        { "PINYIN", "collation", "pinyin" }, /* Solaris variant */
-        { "STROKE", "collation", "stroke" }  /* Solaris variant */
+        { "art_LOJBAN",     "jbo" }, /* registered name */
+        { "cel_GAULISH",    "cel__GAULISH" }, /* registered name */
```

Comment:
why is this CANONICALIZE_MAP differ from CANONICALIZE_MAP in  icu4c/source/common/uloc.cpp . Please either make them the same or comment about why they are different. (and what are different).

Reply:
I noticed they are different but I don't know the history.
We should have a ticket for reading these mappings from data files.
For this ticket I want to strictly remove the obsolete mappings.